src/Domain.LinnApps/PurchaseOrders/PurchaseOrderService.cs

```diff
@@ -958,8 +958,20 @@ private void SetDetailFieldsForCreation(PurchaseOrderDetail detail, int newOrder
         {
             foreach (var delivery in deliveries)
             {
-                var updatedDelivery = updatedDeliveries.First(x => x.DeliverySeq == delivery.DeliverySeq);
-                delivery.DateRequested = updatedDelivery.DateRequested;
+                if (delivery.QuantityOutstanding > 0)
+                {
+                    var updatedDelivery = updatedDeliveries.First(x => x.DeliverySeq == delivery.DeliverySeq);
```

Comment:
If I'm reading this right it looks like you are just setting all the prices based off of what the delivery price used to be. How is that going to keep them in sync with any changes to the prices on the order detail?